gtda/diagrams/preprocessing.py

```diff
@@ -219,7 +219,8 @@ def fit(self, X, y=None):
         if self.metric in ['landscape', 'heat', 'betti', 'persistent_image']:
```

Comment:
Move all the conditions on the metric inside the _discretize function (for all transformer using metrics)